lib/travis/build/script/dart.rb

```diff
@@ -230,12 +230,18 @@ def dartanalyzer
           def dartfmt
             args = task[:dartfmt]
 
-            if args.is_a?(String)
-              sh.echo "dartfmt arguments aren't supported.", ansi: :red
-            end
-
-            sh.if package_direct_dependency?('dart_style'), raw: true do
-              sh.raw 'function dartfmt() { pub run dart_style:format "$@"; }'
+            # If specified `-dartfmt: sdk` and there is a dependency on `dart_style` we
+            # will use the SDK version of dart_style to run formatting checks instead of
+            # the custom pinned version.
+            if args != 'sdk'
+              if args.is_a?(String)
```

Comment:
Shouldn't this be `!args.nil?`? We don't want people passing `dartfmt: 12` or whatever.